core/gate.h

```diff
@@ -0,0 +1,62 @@
+#ifndef BESS_CORE_GATE_H
```

Comment:
Also, all other header files in the core directory does not have "CORE" in the header guard. Not sure whether we should, but let's keep them consistent at least for now.